pallene/builtins.lua

```diff
@@ -4,24 +4,45 @@
 -- SPDX-License-Identifier: MIT
 
 local types = require "pallene.types"
+local T = types.T
 
 local builtins = {}
 
-local T = types.T
-
-for lua_name, typ in pairs({
-    ["io.write"]     = T.Function({T.String()}, {}),
-    ["math.sqrt"]    = T.Function({T.Float()}, {T.Float()}),
-    ["string.char"]  = T.Function({T.Integer()}, {T.String()}),
-    ["string.sub"]   = T.Function({T.String(), T.Integer(), T.Integer()}, {T.String()}),
-    ["tofloat"]      = T.Function({T.Integer()}, {T.Float()}),
-    ["type"]         = T.Function({T.Any()}, {T.String()}),
-}) do
-    local pallene_name = string.gsub(lua_name, "%.", "_")
-    builtins[pallene_name] = {
-        name = lua_name,
-        typ = typ,
+builtins.functions = {
+    io_write = {
+        name = "io.write",
+        typ = T.Function({ T.String() }, {})
+    },
+    tofloat = {
+        name = "tofloat",
+        typ = T.Function({ T.Integer() }, { T.Float() }),
+    },
```

Comment:
Do we need this `name` field? I suspect that we could just use key=name and value=type, as in the old builtins table:

    ["tofloat"] = T.Function({ T.Integer() }, { T.Float() })